aws/middleware/middleware_test.go

```diff
@@ -187,3 +188,98 @@ func TestAttemptClockSkewHandler(t *testing.T) {
 		})
 	}
 }
+
+func TestRecursionDetection(t *testing.T) {
+	const lambdaFunc = "AWS_LAMBDA_FUNCTION_NAME"
```

Comment:
**suggestion**: You could define these as internal constants in the middleware and the test can just re-use them

e.g.

```go
// file: recursion_detection.go

const envAwsLambdaFunctionName = "AWS_LAMBDA_FUNCTION_NAME"
const envAmznTraceId = "_X_AMZN_TRACE_ID"
const amznTraceIdHeader = "X-Amzn-Trace-Id"
```